cli/slsa-verifier/verify/verify_image.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2022 SLSA Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package verify
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/slsa-framework/slsa-verifier/options"
+	"github.com/slsa-framework/slsa-verifier/verifiers"
+	"github.com/slsa-framework/slsa-verifier/verifiers/container"
+)
+
+// Note: nil branch, tag, version-tag and builder-id means we ignore them during verification.
+type VerifyImageCommand struct {
+	// May be nil if supplied alongside in the registry
+	ProvenancePath  *string
+	BuilderID       *string
+	Source          string
+	Branch          *string
+	Tag             *string
+	VersionTag      *string
+	Inputs          map[string]string
+	PrintProvenance bool
+}
+
+func (c *VerifyImageCommand) Exec(ctx context.Context, artifacts []string) (string, error) {
+	artifactHash, err := container.GetImageDigest(artifacts[0])
```

Comment:
Let's return an error if more than one artifact is passed? I don't know if this is possible, but just in case, so that we don't silently ignore it.

Reply:
This is already done: it's handled in the cobra code:
https://github.com/slsa-framework/slsa-verifier/blob/1f42a806ae5412bb2d00f6bc6b1e282aebeb436c/cli/slsa-verifier/verify.go#L30-L34

Reply:
Thanks. Maybe just add a comment here then. I don't mind having a redundant check here to make it explicit to readers.